packages/mesh/src/MeshMaterial.js

```diff
@@ -17,19 +17,32 @@ export default class MeshMaterial extends Shader
      * @param {object} [options] - Additional options
      * @param {number} [options.alpha=1] - Default alpha.
      * @param {number} [options.tint=0xFFFFFF] - Default tint.
+     * @param {string} [options.pluginName='batch'] - Renderer plugin for batching.
+     * @param {PIXI.Program} [options.program=0xFFFFFF] - Custom program.
+     * @param {object} [options.uniforms] - Custom uniforms.
      */
     constructor(uSampler, options)
     {
-        const program = Program.from(vertex, fragment);
-
         const uniforms = {
             uSampler,
             alpha: 1,
             uTextureMatrix: Matrix.IDENTITY,
             uColor: new Float32Array([1, 1, 1, 1]),
         };
 
-        super(program, uniforms);
+        // Set defaults
+        options = Object.assign({
+            tint: 0xFFFFFF,
+            alpha: 1,
+            pluginName: 'batch',
+        }, options);
+
+        if (options.uniforms)
```

Comment:
You don't need the `if` check here, doing `Object.assign({}, undefined)` is fine

Reply:
We are checking uniforms here, and I do not want to create those extra objects for them to be overwritten.